### ISSUE TITLE
Remote library import is marked as error if /RPC2 is used inconsistently with red.xml definitions

### ISSUE THREAD

**Issue by nlamblin** [nokia/RED#361]
I use `RemoteLibrary`.

RED say that it is unable to connect with Remote.

![image](https://user-images.githubusercontent.com/17138972/73063368-e969c780-3e9e-11ea-9953-09c43df8327b.png)

However, connection is working fine. I can call Java keywords through the Remote connection.
There is a solution to bypass this please?

Indeed, before each execution, I faced this popup and it is not ideal.

<img width="436" alt="Capture2" src="https://user-images.githubusercontent.com/17138972/73063466-20d87400-3e9f-11ea-8c9f-42c1f0e64963.PNG">

Best regards,

**Comment by michalanglart** [comment]
Hi,

First of all can you find your Remote library under `Robot Standard libraries` in `Project Explorer` and click `Reload` from context menu and show us the message from the error dialog which would appear?

Regarding popup: you can check `Always launch without asking` checkbox to hide it for good.

**Comment by nlamblin** [comment]
Thank you for your anwser.

The message displayed after the reload is the following one:
Library specification generation problem

Problem occurred during generating libdoc for 'Remote [http://127.0.0.1:64494/]' library.
```
Following exception has been thrown:
            Traceback (most recent call last):
              File "C:\...\Temp\RobotTempDir6014563875802195403\robot_session_server.py", line 43, in inner
                result['result'] = func(*args, **kwargs)
              File "C:\...\Temp\RobotTempDir6014563875802195403\robot_session_server.py", line 105, in inner
                return to_call(*args, **kwargs)
              File "C:\...\Temp\RobotTempDir6014563875802195403\robot_session_server.py", line 138, in inner
                return to_call(*args, **kwargs)
              File "C:\...\Temp\RobotTempDir6014563875802195403\robot_session_server.py", line 78, in inner
                return func(*args, **kwargs)
              File "C:\...\Temp\RobotTempDir6014563875802195403\robot_session_server.py", line 363, in create_libdoc
                return red_libraries.create_libdoc(libname, format)
              File "C:\...\Temp\RobotTempDir6014563875802195403\red_libraries.py", line 64, in create_libdoc
                raise Exception(result)
            Exception: Getting keyword names from library 'Remote' failed: Calling dynamic method 'get_keyword_names' failed: Connecting remote server at http://127.0.0.1:64494/ failed: <ProtocolError for 127.0.0.1:64494/: 404 Not Found>
            
            Try --help for usage information.
            
```

**Comment by michalanglart** [comment]
Can you show the entry in your `red.xml` file? For my the exemplary library works however the path written in `red.xml` has to match the one used when importing. In my case I have:

```
    <remoteLocations uri="http://127.0.0.1:64494"/>
```

and:
```
Library    Remote    http://127.0.0.1:64494
```

although it seems there are some issues regarding `/RPC2` path being added automatically

**Comment by nlamblin** [comment]
Of course, please find it below:

<img width="688" alt="Capture" src="https://user-images.githubusercontent.com/17138972/73067139-238b9700-3ea8-11ea-947a-e2b912192aed.PNG">

**Comment by michalanglart** [comment]
And how is you server configured? Under what path have you started your library? Is it `/RPC2`?

**Comment by nlamblin** [comment]
Yes it is `RPC2`. 

The problem seems to be fixed... I just remove the last `/` in `red.xml`. Whqt is your advice? It is better to add `/` in robot script or to remove it in `red.xml`? Thqnk you.

**Comment by michalanglart** [comment]
Adding `/` in robot script will not help. The problem is that you've started your server under in `host:port` under `/RPC2` path, but then in red.xml you're telling that the library is located in `host:port` under `/` path - this causes `404` error when RED tries to generate library specification file using `robot.libdoc`.

Once you remove `/` part in red.xml you're using the property described here: http://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#importing-remote-library in third **Note** (that once there is no path then `/RPC2` is automatically added). That's why RED is then able to connect and generate the specification file.

If you want to have the library under `/` then you need to configure it for that path:

```
server.putLibrary("/", new ExampleLibrary());
```

and then `red.xml` needs to have an entry with `/` and you would also have to add it in the usage places.

**Comment by nlamblin** [comment]
Thank you very much for the explanation !

**Comment by michalanglart** [comment]
I'll keep that open because it seems that there is a minor regression in two cases:

1) if the Remote library is added to red.xml as `host:port` then it has to be imported the same way in suite file. An error is generated if it is imported this way:
```
Library    host:port/RPC2
```
despite the fact it will work ok in Robot

2) symmetrically if library is added as `host:port/RPC2` then it has to be imported the same way in suite file. An error is generated when it is imported like this:
```
Library    host:port
```
but it will run fine in Robot.


I think somewhen in the past it was handled and working ok.

**Comment by michalanglart** [comment]
Fixed in 0.9.4
